### PR TITLE
ACM-6105: update binary name for hcp for mce 2.4

### PR DESCRIPTION
**Pull request by dtthuynh**
As part of https://issues.redhat.com/browse/ACM-6105 https://github.com/stolostron/hypershift-addon-operator/pull/270 changes, hypershift binary packaged with MCE will be called hcp instead. Update tests to reflect this.

### CODE REVIEW

**Comment by dtthuynh** [comment]
cc @LiangquanLi930 @rokej @davidvossel

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.14-periodics-e2e-aws-ovn-mce-conformance

**Comment by LiangquanLi930** [comment]
/pj-rehearse periodic-ci-openshift-hypershift-release-4.13-periodics-e2e-aws-ovn-mce-conformance

File: ci-operator/step-registry/hypershift/mce/aws/destroy/hypershift-mce-aws-destroy-chain.yaml [hunk 1]
```diff
@@ -13,24 +13,31 @@ chain:
...
+          cd /tmp && tar -xvf /tmp/${HYPERSHIFT_NAME}.tar.gz
+          chmod +x /tmp/${HYPERSHIFT_NAME}
           cd -
         fi
 
         CLUSTER_NAME="$(echo -n $PROW_JOB_ID|sha256sum|cut -c-20)"
 
         echo "$(date) dump HyperShift cluster ${CLUSTER_NAME}"
-        /tmp/hypershift dump cluster --artifact-dir=$ARTIFACT_DIR \
+        /tmp/${HYPERSHIFT_NAME} dump cluster --artifact-dir=$ARTIFACT_DIR \
```
[the remark this answers is not in the document]
Comment (LiangquanLi930) [reply ci-operator/step-registry/hypershift/mce/aws/destroy/hypershift-mce-aws-destroy-chain.yaml:34]:
```
hcp --help                                                                   
Usage:
  hcp [flags]
  hcp [command]

Available Commands:
  completion  Generate the autocompletion script for the specified shell
  create      Commands for creating HostedClusters
  destroy     Commands for destroying HostedClusters
  help        Help about any command

Flags:
  -h, --help      help for hcp
  -v, --version   version for hcp

Use "hcp [command] --help" for more information about a command.
```
We need to still use hypershift cli

File: ci-operator/step-registry/hypershift/mce/aws/create/hypershift-mce-aws-create-chain.yaml [hunk 2]
```diff
@@ -42,7 +49,7 @@ chain:
           PULL_SECRET_PATH="/tmp/.dockerconfigjson"
         fi
       
-        /tmp/hypershift create cluster aws ${ICSP_COMMAND} \
+        /tmp/${HYPERSHIFT_NAME} create cluster aws ${ICSP_COMMAND} \
```
[the remark this answers is not in the document]
Comment (dtthuynh) [reply ci-operator/step-registry/hypershift/mce/aws/create/hypershift-mce-aws-create-chain.yaml:52]: Thanks @davidvossel, really good to know... does this only affect kubevirt or all platforms? (I noticed the bug only specified connectivity issue with kubevirt example)

**Comment by dtthuynh** [comment]
separating to another pr https://github.com/openshift/release/pull/42938

**Pull request closed without merging**